### PR TITLE
Action diary feature v2

### PR DESCRIPTION
**Pull request by LBHSPreston**
Created a new version of the action diary endpoint.  This requires less input parameters and also updates the Action Diary record with the detail of the recording user rather than the API user id.

### CODE REVIEW

File: LBHTenancyAPI/Infrastructure/V2/UseCase/Execution/ExecuteWrapper.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                else
+                {
+                    IsSuccess = true;
+                    Result = response;
+                }
+                    
+            }
+            else
+            {
+                IsSuccess = true;
```
Comment (cob16) [inline LBHTenancyAPI/Infrastructure/V2/UseCase/Execution/ExecuteWrapper.cs]: is this required

File: LBHTenancyAPI/UseCases/V2/ArrearsActions/CreateArrearsActionDiaryUseCase.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+namespace LBHTenancyAPI.UseCases.V2.ArrearsActions
+{
+    /// <summary>
+    /// Use Case for creating Arrears actions diary entry
+    /// </summary>
+    public class CreateArrearsActionDiaryUseCase: ICreateArrearsActionDiaryUseCase
+    {
+        private readonly IArrearsActionDiaryGateway _arrearsActionDiaryGateway;
+        public readonly IArrearsServiceRequestBuilder _requestBuilder;
```
Comment (cob16) [inline LBHTenancyAPI/UseCases/V2/ArrearsActions/CreateArrearsActionDiaryUseCase.cs]: Should not have a _

File: LBHTenancyAPITest/Test/UseCases/V2/Arrears/CreateArrearsActionDiaryUseCaseTests.cs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            //arrange
+            var tenancyAgreementRef = "Test";
+            _fakeGateway.Setup(s => s.CreateActionDiaryEntryAsync(It.Is<ArrearsActionCreateRequest>(i=> i.ArrearsAction.TenancyAgreementRef.Equals("Test"))))
+                .ReturnsAsync(new ArrearsActionResponse
+                {
+                    Success = true,
+                    ArrearsAction = new ArrearsActionLogDto
+                    {
+                        Id = 1,
+                        TenancyAgreementRef = tenancyAgreementRef
```
Comment (cob16) [inline LBHTenancyAPITest/Test/UseCases/V2/Arrears/CreateArrearsActionDiaryUseCaseTests.cs]: This should test all attr that are passed to it

File: LBHTenancyAPI/Controllers/V2/ArrearsActionDiaryController.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        /// <summary>
+        /// Arrears Action Diary Controller V2 to create an action diary record with the specified user recorded
+        /// </summary>
+        /// <param name="actionBalance"></param>
+        /// <param name="actionCategory"></param>
+        /// <param name="actionCode"></param>
+        /// <param name="comment"></param>
+        /// <param name="tenancyAgreementRef"></param>
+        /// <param name="companyCode"></param>
+        /// <param name="appUser"></param>
```
Comment (cob16) [inline LBHTenancyAPI/Controllers/V2/ArrearsActionDiaryController.cs]: consider changing param name to `username` as this no longer needs to be a "app" but rather a real user

File: LBHTenancyAPI/UseCases/V2/ArrearsActions/CreateArrearsActionDiaryUseCase.cs (hunk 2 above)
Comment (LBHSPreston) [reply LBHTenancyAPI/UseCases/V2/ArrearsActions/CreateArrearsActionDiaryUseCase.cs]: The interface declaration should be private.  Changed to private.

File: LBHTenancyAPI/Controllers/V2/ArrearsActionDiaryController.cs (hunk 4 above)
Comment (LBHSPreston) [reply LBHTenancyAPI/Controllers/V2/ArrearsActionDiaryController.cs]: This is now done.

File: LBHTenancyAPI/Infrastructure/V2/UseCase/Execution/ExecuteWrapper.cs (hunk 1 above)
Comment (LBHSPreston) [reply LBHTenancyAPI/Infrastructure/V2/UseCase/Execution/ExecuteWrapper.cs]: It doesn't look like this is required.  Appears to be redundant.  I've removed it and all checks continue to pass.

File: LBHTenancyAPITest/Test/UseCases/V2/Arrears/CreateArrearsActionDiaryUseCaseTests.cs (hunk 3 above)
Comment (LBHSPreston) [reply LBHTenancyAPITest/Test/UseCases/V2/Arrears/CreateArrearsActionDiaryUseCaseTests.cs]: Added remaining attributes to test.

**Pull request merged**